### PR TITLE
Add pip packaging

### PR DESCRIPTION
**Pull request by vemonet**
For https://github.com/uav4geo/LibreTranslate/issues/30

Make sure to use python3.8 to install:

```bash
python3.8 -m venv .venv
source .venv/bin/activate
```

Install locally for dev:

```bash
pip install -e .
```

Run:

```bash
libretranslate
```

* First time download models with Argos

* Any other time directly starts the API

With args:

```bash
libretranslate --port 8081
```

### What I did

Added pip setup package requirements: 

* add  `setup.py` (needs to be completed by @pierotofy )
* Added `app/__init__.py` to make it a proper pip package
* Moved `main.py` to `app` folder so that it is included in the pip package
* Use `pkg_resource` to get templates folder data
* Updated the Dockerfile (did not tried it though, we will see the github action workflows outputs!)
* Updated the README

In "check and install model" got an uncompatibility with argos to check package `package.get_installed_packages()` returned 1 even if nothing installed

```python
if len(package.get_installed_packages()) == 0 or force:
# replaced by
if len(package.get_installed_packages()) < 2 or force:
```

### Note about compatibility with 3.6 and 3.7

@PJ-Finlay it seems like I am still running into issues when running with 3.6 and 3.7

```bash
python3 main.py                                           
Moving Argos Translate data dir from /home/vemonet/.argos-translate/packages to /home/vemonet/.local/share/argos-translate/packages
Traceback (most recent call last):
  File "main.py", line 39, in <module>
    frontend_timeout=args.frontend_timeout)
  File "/home/vemonet/develop/perso/LibreTranslate/app/app.py", line 17, in create_app
    from app.init import boot
  File "/home/vemonet/develop/perso/LibreTranslate/app/init.py", line 3, in <module>
    from argostranslate import settings, package, translate
  File "/home/vemonet/.local/lib/python3.6/site-packages/argostranslate/settings.py", line 39, in <module>
    shutil.copytree(legacy_package_data_dir, package_data_dir)
  File "/usr/lib/python3.6/shutil.py", line 321, in copytree
    os.makedirs(dst)
  File "/usr/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/home/vemonet/.local/share/argos-translate/packages'
```

### What needs to be done

* @pierotofy  needs to go over the `setup.py` script and README
* I would like to add a simple test in the workflow to at least check the initialization with argos works well (then we can expand from there)
* Improve the integration of the new package download from argos, it seems to still have some hiccups 

### Finish setup of workflow to publish to pypi

I added a GitHub Actions workflow to automatically publish to pypi when a release is created on GitHub (you can also manually trigger it if the auto run failed)

You will need to define those 2 secrets to make it work: `PYPI_USERNAME` and `PYPI_PASSWORD`

I added a job to also run `pytest` with multiple version of python to test the package before publishing

I created a `tests` folder with `__init__.py` to get started

All we need now is to add some classic python tests files in this folder and uncomment the testing part

### Notes for later

* Do not put regular emojis in the `README.md` last time I did there was some issues with pypi not able to parse the README
* If you add non python files: add them in a folder in your `app/` folder (like for templates), then in the code use `pkg_resource.package_filename` to get the file

Apparently it works out of the box with your render_template:

```python
@app.route("/")
    def index():
        return render_template('index.html', gaId=ga_id, frontendTimeout=frontend_timeout)
```

But you might want to make sure it works properly for everyone

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    },
+
+    python_requires='>=3.6.0',
+    long_description=open('README.md').read(),
+    long_description_content_type="text/markdown",
+    install_requires=open("requirements.txt", "r").readlines(),
+    tests_require=['pytest==5.2.0'],
+    setup_requires=['pytest-runner'],
+    classifiers=[
+        "License :: OSI Approved :: MIT License",
```
Comment (PJ-Finlay) [inline setup.py:31]: AGPL?

**Comment by PJ-Finlay** [comment]
The error you are getting with Python 3.6 is because it's trying to copy `~/.argos-translate/packages` (the legacy data dir) to ~/.local/share/argos-translate/packages` which already exists. `rm -rf ~/.argos-translate` should fix the issue and `rm -rf ~/.local/share/argos-translate` will clear all packages and the package index.

File: setup.py (hunk 1 above)
Comment (vemonet) [reply setup.py:31]: Fixed

**Comment by vemonet** [comment]
I added a workflow and quick test for `boot()` which download the packages and check if argos find them

It's nice to test the complete process, but @PJ-Finlay you might want to avoid putting more pressure on the services providing the models for download 

Here is a run passing: https://github.com/vemonet/LibreTranslate/actions/runs/551614916

**Comment by pierotofy** [comment]
This is so awesome! :clap: 

Pip package is up: https://pypi.org/project/libretranslate/

**Comment by pierotofy** [comment]
Thanks @vemonet !!

**Pull request merged**